src/grpc/users.rs

```diff
@@ -0,0 +1,363 @@
+        log_received!(&request);
+        let token = tonic_auth!(
+            get_token_from_md(request.metadata()),
+            "Token authentication error"
+        );
+        let request = DeactivateUser(request.into_inner());
+        let ctx = Context::admin();
+        tonic_auth!(
+            self.authorizer.check_permissions(&token, vec![ctx]).await,
+            "Unauthorized"
+        );
+        let (user_id, user) = tonic_internal!(
+            self.database_handler.deactivate_user(request).await,
+            "Internal deactivate user error"
+        );
+        self.cache.update_user(&user_id, user);
+
+        return_with_log!(DeactivateUserResponse {});
+    }
+
+    async fn activate_user(
+        &self,
+        request: Request<ActivateUserRequest>,
+    ) -> Result<Response<ActivateUserResponse>, Status> {
+        log_received!(&request);
+        let token = tonic_auth!(
+            get_token_from_md(request.metadata()),
+            "Token authentication error"
+        );
+        let request = ActivateUser(request.into_inner());
+        let ctx = Context::admin();
+        tonic_auth!(
+            self.authorizer.check_permissions(&token, vec![ctx]).await,
+            "Unauthorized"
+        );
+        let (user_id, user) = tonic_internal!(
+            self.database_handler.activate_user(request).await,
+            "Internal deactivate user error"
+        );
+        self.cache.update_user(&user_id, user);
+
+        return_with_log!(ActivateUserResponse {});
+    }
+
+    async fn create_api_token(
+        &self,
+        _request: Request<CreateApiTokenRequest>,
+    ) -> Result<Response<CreateApiTokenResponse>, Status> {
+        todo!()
+    }
+
+    async fn get_api_token(
+        &self,
+        request: Request<GetApiTokenRequest>,
+    ) -> Result<Response<GetApiTokenResponse>, Status> {
+        log_received!(&request);
+        let token = tonic_auth!(
+            get_token_from_md(request.metadata()),
+            "Token authentication error"
+        );
+        let request = GetToken(request.into_inner());
+        let ctx = Context::self_ctx();
+        let user_id = tonic_auth!(
+            self.authorizer.check_permissions(&token, vec![ctx]).await,
+            "Unauthorized"
+        )
+        .ok_or_else(|| Status::internal("User id not returned"))?;
+        let user = tonic_invalid!(
+            self.cache
+                .get_user(&user_id)
+                .ok_or_else(|| anyhow!("Not found")),
+            "User not found"
+        );
+        let token_id = tonic_invalid!(request.get_token_id(), "Invalid token_id");
+        let token = match user.attributes.0.tokens.get(&token_id) {
+            Some(token) => Some(into_api_token(token_id, token.clone())),
+            None => return Err(Status::not_found("Token not found")),
+        };
+        let response = GetApiTokenResponse { token };
+
+        return_with_log!(response);
+    }
+
+    async fn get_api_tokens(
```

Comment:
Shouldn´t this be easy to do as-well ? Since the tokens are now part of a user this should be quite straight forward.